test/functional/feature_coinstatsindex.py

```diff
@@ -0,0 +1,313 @@
+                }
+            })
+            self.block_sanity_check(res5['block_info'])
+
+        # Generate and send a normal tx with two outputs
+        tx1_inputs = []
+        tx1_outputs = {self.nodes[0].getnewaddress(): 21, self.nodes[0].getnewaddress(): 42}
+        raw_tx1 = self.nodes[0].createrawtransaction(tx1_inputs, tx1_outputs)
+        funded_tx1 = self.nodes[0].fundrawtransaction(raw_tx1)
+        signed_tx1 = self.nodes[0].signrawtransactionwithwallet(funded_tx1['hex'])
+        tx1_txid = self.nodes[0].sendrawtransaction(signed_tx1['hex'])
+
+        # Find the right position of the 21 BTC output
+        tx1_final = self.nodes[0].gettransaction(tx1_txid)
+        for output in tx1_final['details']:
+            if output['amount'] == Decimal('21.00000000') and output['category'] == 'receive':
+                n = output['vout']
+
+        # Generate and send another tx with an OP_RETURN output (which is unspendable)
+        tx2 = CTransaction()
+        tx2.vin.append(CTxIn(COutPoint(int(tx1_txid, 16), n), b''))
+        tx2.vout.append(CTxOut(int(20.99 * COIN), CScript([OP_RETURN] + [OP_FALSE]*30)))
+        tx2_hex = self.nodes[0].signrawtransactionwithwallet(ToHex(tx2))['hex']
+        self.nodes[0].sendrawtransaction(tx2_hex)
+
+        # Include both txs in a block
+        self.nodes[0].generate(1)
+        self.sync_all()
+
+        self.wait_until(lambda: not try_rpc(-32603, "Unable to read UTXO set", index_node.gettxoutsetinfo, 'muhash'))
+        for hash_option in index_hash_options:
+            # Check all amounts were registered correctly
+            res6 = index_node.gettxoutsetinfo(hash_option, 108)
+            assert_equal(res6['total_unspendable_amount'], Decimal('70.98999999'))
+            assert_equal(res6['block_info'], {
+                'unspendable': Decimal('20.98999999'),
+                'prevout_spent': 111,
+                'new_outputs_ex_coinbase': Decimal('89.99993620'),
+                'coinbase': Decimal('50.01006381'),
+                'unspendables': {
+                    'genesis_block': 0,
+                    'bip30': 0,
+                    'scripts': Decimal('20.98999999'),
+                    'unclaimed_rewards': 0
+                }
+            })
+            self.block_sanity_check(res6['block_info'])
+
+        # Create a coinbase that does not claim full subsidy and also
+        # has two outputs
+        cb = create_coinbase(109, nValue=35)
+        cb.vout.append(CTxOut(5 * COIN, CScript([OP_FALSE])))
+        cb.rehash()
+
+        # Generate a block that includes previous coinbase
+        tip = self.nodes[0].getbestblockhash()
+        block_time = self.nodes[0].getblock(tip)['time'] + 1
+        block = create_block(int(tip, 16), cb, block_time)
+        block.solve()
+        self.nodes[0].submitblock(ToHex(block))
+        self.sync_all()
+
+        self.wait_until(lambda: not try_rpc(-32603, "Unable to read UTXO set", index_node.gettxoutsetinfo, 'muhash'))
```

Comment:
why is this needed? (sync_all should call syncwithvalidationinterface). Also, if it didn't, shouldn't the index RPC call syncwithvalidationinterface internally by itself, similar to how the wallet does it?

Reply:
True, I missed the option to use the `BlockUntilSynced...` function from BaseIndex.